posthog/tasks/alerts/checks.py

```diff
@@ -217,6 +221,35 @@ def check_alert(alert_id: str) -> None:
         )
         return
 
+    if alert.is_calculating:
+        logger.warning(
+            "Alert is already being computed so skipping checking it now",
+            alert=alert,
+        )
+        return
+
+    alert.is_calculating = True
+    alert.save()
+
+    try:
+        check_alert_atomically(alert)
+    except Exception:
+        raise
+    finally:
+        # Get all updates with alert checks
+        alert.refresh_from_db()
+        alert.is_calculating = False
```

Comment:
I think eventually we'll have to treat alerts with is_calculating set to True but their last update date some amount of time in the past the same as errored alerts, but otherwise looks good for now!